srml/staking/src/lib.rs

```diff
@@ -16,7 +16,157 @@
+//! 
+//! ## Interface
+//! 
+//! ### Types
+//! 
+//! - `Currency`: Used as the measurement means of staking and funds management.		
+//!  
+//! ### Dispatchable
+//! 
+//! The Dispatchable functions of the staking module enable the steps needed for entities to accept and change their role, alongside some helper functions to get/set the metadata of the module.
+//! 
+//! Please refer to the [`Call`] enum and its associated variants for a detailed list of dispatchable functions.
+//! 
+//! ### Public 
+//! The staking module contains many public storage items and (im)mutable functions. Please refer to the [struct list](#structs) below and the [`Module`](https://crates.parity.io/srml_staking/struct.Module.html) struct definition for more details.
+//! 
+//! ## Usage
+//! 
+//! 
+//! ### Snippet: Bonding and Accepting Roles
+//! 
+//! An arbitrary account pair, given that the associated stash has the required funds, can become stakers via the following call:
+//! 
+//! ```rust,ignore
+//! // bond account 3 as stash
+//! // account 4 as controller 
+//! // with stash value 1500 units 
+//! // while the rewards get transferred to the controller account.
+//! Staking::bond(Origin::signed(3), 4, 1500, RewardDestination::Controller);
+//! ```
+//! 
+//! To state desire in becoming a validator: 
+//! 
+//! ```rust,ignore
+//! // controller account 4 states desire for validation with the given preferences.
+//! Staking::validate(Origin::signed(4), ValidatorPrefs::default()); 
+//! ```
+//! 
+//! Note that, as mentioned, the stash account is transparent in such calls and only the controller initiates the function calls.
+//! 
+//! Similarly, to state desire in nominating: 
+//! 
+//! ```rust,ignore
+//! // controller account 4 nominates for account 10 and 20.
+//! Staking::nominate(Origin::signed(4), vec![20, 10]);
+//! ```
+//! 
+//! Finally, account 4 can withdraw from any of the above roles via
+//! 
+//! ```rust,ignore
+//! Staking::chill(Origin::signed(4));
+//! ```
+//! 
+//! ## Implementation Details
+//! 
+//! ### Slot Stake 
+//! 
+//! The term `slot_stake` will be used throughout this section. It refers to a value calculated at the end of each era, containing the _minimum value at stake among all validators._
+//! 
+//! ### Reward Calculation 
+//! 
+//! - Rewards are recorded **per-session** and paid **per-era**. The value of reward for each session is calculated at the end of the session based on the timeliness of the session, then accumulated to be paid later. The value of the new _per-session-reward_ is calculated at the end of each era by multiplying `slot_stake` and a configuration storage named `SessionReward`. 
```

Comment:
* The value of **the** reward for each session
* configuration storage **item** named